string motion
string object
string optional_target

